### PR TITLE
tp delivery

### PR DESCRIPTION
**Pull request by n1tr0-5urf3r**
Added tp delivery command

### CODE REVIEW

File: guildwars2/commerce.py [hunk 1]
```diff
@@ -142,6 +142,72 @@ class CommerceMixin:
...
+    @tp.command(name="delivery")
+    @commands.cooldown(1, 10, BucketType.user)
+    async def tp_delivery(self, ctx):
+        """Show items and coins delivered to 
+            tradingpost for you
+            
+        Required permissions: tradingpost
+        """
+        user = ctx.author
+        #language = self.getlanguage(ctx)
```
Comment (Maselkov) [inline guildwars2/commerce.py]: No need for this

File: guildwars2/commerce.py [hunk 2]
```diff
@@ -142,6 +142,72 @@ class CommerceMixin:
...
+
+        coins = results["coins"]
+        items = results["items"]
+        items = items[:20]  # Get only first 20 entries
+        item_quantity = []
+        itemlist = []
+
+        # Get coins
+        if coins is 0:
+            gold = "No monnies for you"
```
Comment (Maselkov) [inline guildwars2/commerce.py]: probably should change that

File: guildwars2/commerce.py [hunk 3]
```diff
@@ -142,6 +142,72 @@ class CommerceMixin:
...
+                item_quantity.append(str(item["count"]))
+                itemdoc = await self.fetch_item(item["id"])
+                itemlist.append(itemdoc)
+
+            for item in itemlist:
+                item_name = item["name"]
+                # Get quantity of items
+                quantity = item_quantity[counter]
+                counter += 1
+                data.add_field(name=item_name, value=str(quantity) + 'x', inline=False)
```
Comment (Maselkov) [inline guildwars2/commerce.py]: str.format() would probably work nicer

File: guildwars2/commerce.py [hunk 4]
```diff
@@ -142,6 +142,72 @@ class CommerceMixin:
...
+        else:
+            data.add_field(name="No current deliveries.", value="Have fun!", inline=False)
+
+        try:
+            await ctx.send(embed=data)
+        except discord.HTTPException as e:
+            await ctx.send("Need permission to embed links " + str(e))
```
Comment (Maselkov) [inline guildwars2/commerce.py]: don't want to expose exceptions to user

File: guildwars2/guild/general.py [hunk 5]
```diff
@@ -159,3 +160,58 @@ class GeneralGuild:
...
+                        multiplier = ""
+                    else:
+                        itemdoc = await self.fetch_item(entry["item_id"])
+                        item_name = itemdoc["name"]
+                        multiplier = "x "
+                    if entry["operation"] is "withdraw":
+                        operator = " withdrew "
+                    else:
+                        operator = " deposited "
+                    data.add_field(name=timedate, value=user + operator + str(quantity) + multiplier + item_name,
```
Comment (Maselkov) [inline guildwars2/guild/general.py]: str.format() would work nicer

File: guildwars2/guild/general.py [hunk 6]
```diff
@@ -159,3 +160,58 @@ class GeneralGuild:
...
+                                   inline=False)
+                    counter += 1
+
+        try:
+            await ctx.send(embed=data)
+        except discord.Forbidden:
+            await ctx.send("Need permission to embed links")
```
Comment (Maselkov) [inline guildwars2/guild/general.py]: no newline :c

File: guildwars2/commerce.py (hunk 1 above)
Comment (n1tr0-5urf3r) [reply guildwars2/commerce.py]: removed the language part

File: guildwars2/commerce.py (hunk 2 above)
Comment (n1tr0-5urf3r) [reply guildwars2/commerce.py]: Changed to "Currently no coins for pickup"

File: guildwars2/commerce.py (hunk 3 above)
Comment (n1tr0-5urf3r) [reply guildwars2/commerce.py]: changed to str.format(quantity)

File: guildwars2/guild/general.py (hunk 5 above)
Comment (n1tr0-5urf3r) [reply guildwars2/guild/general.py]: changed to str.format(quantity)

File: guildwars2/guild/general.py (hunk 6 above)
Comment (n1tr0-5urf3r) [reply guildwars2/guild/general.py]: added newline

File: guildwars2/commerce.py (hunk 4 above)
Comment (n1tr0-5urf3r) [reply guildwars2/commerce.py]: removed exception in output

File: guildwars2/commerce.py [hunk 7]
```diff
@@ -198,14 +197,14 @@ class CommerceMixin:
                 # Get quantity of items
                 quantity = item_quantity[counter]
                 counter += 1
-                data.add_field(name=item_name, value=str(quantity) + 'x', inline=False)
+                data.add_field(name=item_name, value=str.format(quantity) + 'x', inline=False)
```
Comment (Maselkov) [inline guildwars2/commerce.py]: That's not what I meant :P
```py
"{}x".format(quantity)
```

**Comment by Maselkov** [comment]
Needs some cleanup, but good work

**Pull request merged**